w3-blob.md

```diff
@@ -0,0 +1,199 @@
+type LocationClaim struct {
+  content    Multihash
+  url        URL
+  range      ByteRange
+}
+
+type ByteRange struct {
+  offset    Int
+  length    Int
+} representation tuple
+```
+
+#### Add Blob Success
+
+##### Add Blob Allocated
+
+Capability provider MUST issue receipt with `BlobAddAllocation` success result if underlying space has enough capacity for the requested blob, but has not such blob locally.
+
+The `url` field MUST be set to the HTTP PUT endpoint where `content` matching specified [multihash] and `size` could be uploaded.
+
+The `headers` field MUST be set to the HTTP headers to be send along with `content` to the HTTP PUT endpoint.
+
+###### Add Blob Committed
+
+Capability provider MUST issue receipt with `LocationClaim` capability delegated to the [space] when requested blob is added to the space, which can happen if provider already has a matching blob locally.
+
+By delegating `LocationClaim` to the space provider makes a commitment to serve corresponding blob at byte range in from given URL to all authorized agents.
+
+> Not that since `aud` of the claim is the `space` DID, space can authorize other principals by re-delegating this claim.
+
+## Get Blob
+
+Authorized agent MAY invoke `/space/content/get/blob` capability on the [space] subject to query a state of the corresponding blob.
+
+### Get Blob Capability
+
+#### Get Blob Capability Schema
+
+```ipldsch
+type struct GetBlob {
+  cmd   "/space/content/get/blob"
+  sub   SpaceDID
+  args  BlobQuery
+}
+
+type struct BlobQuery {
+  content   Multihash
+}
+```
+
+### Get Blob Receipt
+
+#### Get Blob Result
+
+```ipldsch
+type BlobAddResult union {
+  BlobGetSuccess    "ok"
+  BlobGetFailure    "error"
+} representation keyed
+
+type BlobGetSuccess union {
+  BlobAddAllocation     "allocated"
+  &UCANLocationClaim    "committed"
+} representation keyed
+
+type BlobGetFailure struct {
+  message     string
+}
+```
+
+#### Get Blob Success
+
+Capability provider MUST return the state of the requested blob. If memory for the blob was allocated but content has not been yet uploaded result MUST be of `BlobAddAllocation` case. If blob has been allocated and uploaded result MUST of `UCANLocationClaim` case.
+
+> Please note that implementation MAY check local state on invocation and update from `BlobAddAllocation` to `UCANLocationClaim`, however this SHOULD be done in way that preserves [referential transparency]. In other words no side effects should be observable, instead function should lazily compute and cache.
+
+## Publish Blob
+
+Blob can be published by authorizing read interface (e.g. IPFS gateway) by re-delegating `LocationClaim` for a corresponding blob to it.
+
+> Note that same applies to publishing blob on [IPNI], new capability is not necessary, user simply needs to re-delegate `LocationClaim` to the DID representing [IPNI] publisher. [IPNI] publisher in turn may publish delegation to DID with publicly known private key allowing anyone to perform the reads.
```

Comment:
So once something is published to IPNI, the provider (i.e. W3S) needs to keep updating it without having the user redelegate every time. It's not clear if this flow supports that.

Reply:
Please bare with me as this spec is by no means done and things are not very clear right now. To answer your question however, yes publisher can keep publishing advertisements as long as delegation is valid. If it had been time limited that is in fact was a user intention.